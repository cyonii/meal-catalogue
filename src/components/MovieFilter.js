import { useSelector, useDispatch } from 'react-redux';
import { setFilter } from '../redux/actions';

const MovieFilter = () => {
  const filter = useSelector((state) => state.filter);
  const dispatch = useDispatch();
  const filterMatrix = [
    ['popular', 'Popular'],
    ['top_rated', 'Top Rated'],
    ['upcoming', 'Upcoming'],
    ['now_playing', 'Now Playing'],
    ['latest', 'Latest'],
  ];
  const filterOptions = filterMatrix.map((f) => <option key={f[0]} value={f[0]}>{f[1]}</option>);

  const handleChange = (e) => {
    const { value } = e.target;
    dispatch(setFilter(value));
  };

  return (
    <label className="d-flex align-items-center form-label mb-0" htmlFor="movie-filter">
      Showing
      <select
        value={filter}
        onChange={handleChange}
        className="form-control form-select border-dark w-auto ms-2"
        id="movie-filter"
      >
        {filterOptions}
      </select>
    </label>
  );
};

export default MovieFilter;
