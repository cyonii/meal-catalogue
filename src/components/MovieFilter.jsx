import { useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";
import { setFilter } from "../redux/actions/metaData";

const MovieFilter = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const optionGroups = [
    ["popular", "Popular"],
    ["topRated", "Top Rated"],
    ["upcoming", "Upcoming"],
    ["nowPlaying", "Now Playing"],
  ];
  const filterOptions = optionGroups.map((opGroup) => (
    <option key={opGroup[0]} value={opGroup[0]}>
      {opGroup[1]}
    </option>
  ));

  const handleChange = (e) => {
    dispatch(setFilter(e.target.value));
    history.push("/");
  };

  return (
    <form data-testid="filter-form">
      <label
        className="form-label d-flex align-items-center mb-0"
        htmlFor="filter"
      >
        <span className="hidden lg:inline mr-2 dark:text-slate-100">
          Showing
        </span>
        <select
          onChange={handleChange}
          className=""
          id="filter"
          aria-label="'Showing' filter form"
        >
          {filterOptions}
        </select>
      </label>
    </form>
  );
};

export default MovieFilter;
