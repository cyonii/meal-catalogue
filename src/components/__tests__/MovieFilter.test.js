import { render, screen } from '@testing-library/react';
import MovieFilter from '../MovieFilter';

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
}));

describe('MovieFilter', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<MovieFilter />);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('filter-form')).toBeInTheDocument();
    expect(screen.getByText('Showing')).toBeInTheDocument();
    expect(screen.getByText('Popular')).toBeInTheDocument();
  });
});
