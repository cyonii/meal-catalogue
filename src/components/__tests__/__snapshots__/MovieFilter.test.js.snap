// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieFilter should render correctly 1`] = `
<DocumentFragment>
  <form
    class=""
    data-testid="filter-form"
  >
    <label
      class="form-label d-flex align-items-center mb-0 form-label"
      for="filter"
    >
      <span
        class="d-none d-lg-inline me-2"
      >
        Showing
      </span>
      <select
        aria-label="'Showing' filter form"
        class="form-control form-select border-0 rounded-0 form-select"
        id="filter"
      >
        <option
          value="popular"
        >
          Popular
        </option>
        <option
          value="topRated"
        >
          Top Rated
        </option>
        <option
          value="upcoming"
        >
          Upcoming
        </option>
        <option
          value="nowPlaying"
        >
          Now Playing
        </option>
      </select>
    </label>
  </form>
</DocumentFragment>
`;
